packages/context/src/value-promise.ts

```diff
@@ -218,3 +217,42 @@ export function tryWithFinally<T>(
   }
   return result;
 }
+
+/**
+ * Resolve an iterator of source values into a result until the evaluator
+ * returns `true`
+ * @param source The iterator of source values
+ * @param resolver The resolve function that maps the source value to a result
+ * @param evaluator The evaluate function that decides when to stop
+ */
+export function resolveUntil<T, V>(
```

Comment:
I find the name `resolveUntil` confusing. In my mind, we _resolve_ values from a context (IoC container), but there is no context involved here. I am proposed to rename this method to `mapUntil`.

Reply:
I choose `resolveUntil` to be consistent with other methods such as `resolveList`.

Reply:
Fair enough 👍